### PR TITLE
Us112847/api documentation

### PR DESCRIPTION
**Pull request by TomMichielsD2L**

### CODE REVIEW

File: ApiDocs.md [hunk 1]
```diff
@@ -0,0 +1,859 @@
...
+- 200 OK – Action succeeded
+- 403 Forbidden
+- 404 Not Found
+### API Versions
+- 1.0 - Route first appeared in LMS v20.20.3
+
+# Pending
+
+## DELETE /d2l/api/customization/cpd/(version)/pending/dismiss/(awardId)
+Restore a pending record
```
Comment (gravindran-d2l) [inline ApiDocs.md:266]: Is this meant to restore a "dismissed" record?  Also, in the description you might want to explain that the action of this call deletes a particular status, resulting in a change to the status.

File: ApiDocs.md [hunk 2]
```diff
@@ -0,0 +1,859 @@
...
+Retrieve the list of methods
+### Parameters
+- version (D2LVERSION) – API version
+### Status Codes
+- 200 OK – Action succeeded
+- 403 Forbidden
+### API Versions
+- 1.0 - Route first appeared in LMS v20.19.12
+### Return
+- This action returns a JSON array of Method data blocks
```
Comment (TomMichielsD2L) [inline ApiDocs.md:205]: I just figured out how to link to another section in this markdown document. Should I add that for all the types?

File: ApiDocs.md [hunk 3]
```diff
@@ -0,0 +1,859 @@
...
+### Status Codes
+- 200 OK – Action succeeded
+- 400 Bad Request
+- 403 Forbidden
+### API Versions
+- 1.0 - Route first appeared in LMS v20.19.12
+### Return
+- This action returns a Record JSON block for the newly created record
+### Input
+The record and file data should be provided as formdata. Allowing for multiple files to be uploaded given the filename
```
Comment (gravindran-d2l) [inline ApiDocs.md:469]: "Allowing for multiple files to be uploaded given the filename" is not coherent.

File: ApiDocs.md [hunk 4]
```diff
@@ -0,0 +1,859 @@
...
+### Status Codes
+- 200 OK – Action succeeded
+- 400 Bad Request
+- 403 Forbidden
+### API Versions
+- 1.0 - Route first appeared in LMS v20.20.1
+### Return
+- This action returns a Record JSON block for the newly created record
+### Input
+The record and file data should be provided as formdata. Allowing for multiple files to be uploaded given the filename
```
Comment (gravindran-d2l) [inline ApiDocs.md:488]: Same comment as above.

File: ApiDocs.md [hunk 5]
```diff
@@ -0,0 +1,859 @@
...
+- deletedFiles ([number]) - Optional. list of deleted file ids
+### Status Codes
+- 200 OK – Action succeeded
+- 400 Bad Request
+- 403 Forbidden
+- 404 Not Found
+### API Versions
+- 1.0 - Route first appeared in LMS v20.19.12
+### Input
+The record, file data, and deletedFiles should be provided as formdata. Allowing for multiple files to be uploaded given the filename
```
Comment (gravindran-d2l) [inline ApiDocs.md:507]: As above.

File: ApiDocs.md [hunk 6]
```diff
@@ -0,0 +1,859 @@
...
+## GET /d2l/api/customization/cpd/(version)/report/user/(userId)
+Get a user's record report
+### Parameters
+- version (D2LVERSION) – API version
+- userId (number) - User Id
+### Query Parameters
+- method (Number) - Optional. Method to look for
+- recordName (String) - Optional. Record Name to look for
+- subject (Number) - Optional. Subject to look for
+- startDate (UTCDateTime) -Optional. Retrieve records with a date time that is greater than or equal this
```
Comment (gravindran-d2l) [inline ApiDocs.md:539]: Spaces between hyphen and "Optional"

File: ApiDocs.md [hunk 7]
```diff
@@ -0,0 +1,859 @@
...
+## GET /d2l/api/customization/cpd/(version)/report/csv/user/(userId)
+Get a user's csv record report
+### Parameters
+- version (D2LVERSION) – API version
+- userId (number) - User Id
+### Query Parameters
+- method (Number) - Optional. Method to look for
+- recordName (String) - Optional. Record Name to look for
+- subject (Number) - Optional. Subject to look for
+- startDate (UTCDateTime) -Optional. Retrieve records with a date time that is greater than or equal this
```
Comment (gravindran-d2l) [inline ApiDocs.md:578]: Spaces between hyphen and Optional.

File: ApiDocs.md [hunk 8]
```diff
@@ -0,0 +1,859 @@
...
+- 1.0 - Route first appeared in LMS v20.19.12
+### Return
+- This action returns a Subject JSON block or null
+
+## POST /d2l/api/customization/cpd/(version)/subject
+Create a new Subject
+### Parameters
+- version (D2LVERSION) – API version
+### JSON Parameters
+- SubjectSubjectPostData - data for a new subject
```
Comment (gravindran-d2l) [inline ApiDocs.md:632]: Is it really "SubjectSubjectPostData"?

Comment (jwong-d2l) [inline ApiDocs.md:632]:
```suggestion
- SubjectPostData - data for a new subject
```

Comment (jwong-d2l) [inline ApiDocs.md]:
```suggestion
- SubjectPostData - data for a new subject
```

File: ApiDocs.md [hunk 9]
```diff
@@ -0,0 +1,859 @@
...
+### Return
+- This action returns a Subject JSON block for the newly created subject
+
+## PUT /d2l/api/customization/cpd/(version)/subject/(subjectId)
+Update an existing subject
+### Parameters
+- version (D2LVERSION) – API version
+- subjectId (Number) - Subject ID
+### JSON Parameters
+- SubjectSubjectPostData - data for a new subject
```
Comment (jwong-d2l) [inline ApiDocs.md:647]:
```suggestion
- SubjectMethodPostData - data for a new subject
```

File: ApiDocs.md [hunk 10]
```diff
@@ -0,0 +1,859 @@
...
+- 1.0 - Route first appeared in LMS v20.19.12
+### Return
+- This action returns a Subject JSON block or null
+
+## POST /d2l/api/customization/cpd/(version)/subject
+Create a new Subject
+### Parameters
+- version (D2LVERSION) – API version
+### JSON Parameters
+- SubjectPostData - data for a new subject
```
Comment (jwong-d2l) [inline ApiDocs.md:632]:
```suggestion
- SubjectMethodPostData - data for a new subject
```

File: ApiDocs.md (hunk 1 above)
Comment (TomMichielsD2L) [reply ApiDocs.md:266]: I am having a hard time coming up with something, but I have updated it to `Restore a dismissed record` Do you have a suggestion for what you want this to say?

Comment (gravindran-d2l) [reply ApiDocs.md]: My concern is that when the REST verb is "delete", then the user would expect the subject of that verb to be deleted.  So maybe we want to say that the call removes the "dismissed" status of a record, and returns the record to the "pending" state.

Comment (TomMichielsD2L) [reply ApiDocs.md]: I agree. What about something like "Restore a dismissed record, resetting its status to pending"? I would prefer you just tell me what to type :D

Comment (gravindran-d2l) [reply ApiDocs.md]: How about "Removes the dismissed status of a record, thereby returning the record to a pending state"?

Comment (TomMichielsD2L) [reply ApiDocs.md]: Works for me!

File: ApiDocs.md (hunk 3 above)
Comment (TomMichielsD2L) [reply ApiDocs.md]: Is the new text good for you?
"The record and file data should be provided as formdata; up to 20 files can be uploaded at a time"

Comment (gravindran-d2l) [reply ApiDocs.md]: I like.

**Pull request merged**